### ISSUE TITLE
Test failures on FreeBSD

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#19156]
#19130 fixed a recent compile failure on FreeBSD.

While the compilation failure seems to have been resolved, we still seem to be getting test failures on that platform. E.g. see:

https://ci.buildbot.openssl.org/#/builders/11/builds/9

````
        # INFO:  @ test/bio_dgram_test.c:121
        # # Testing with AF_INET, local=1
        # 
        # ERROR: (bool) 'do_sendmmsg(b1, tx_msg, 2, 0, &num_processed) == true' failed @ test/bio_dgram_test.c:272
        # false
        # 0020E10008000000:error:80000027:system library:dgram_sendmmsg:Destination address required:crypto/bio/bss_dgram.c:1314:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_sendmmsg:local address not available:crypto/bio/bss_dgram.c:1297:
        # 0020E10008000000:error:80000016:system library:dgram_sendmmsg:Invalid argument:crypto/bio/bss_dgram.c:1314:
        # OPENSSL_TEST_RAND_ORDER=1662544797
        not ok 3 - iteration 3
# ------------------------------------------------------------------------------
        # INFO:  @ test/bio_dgram_test.c:127
        # # Testing with AF_INET6, local=1
        # 
        # ERROR: (bool) 'do_recvmmsg(b2, rx_msg, 2, 0, &num_processed) == true' failed @ test/bio_dgram_test.c:323
        # false
        # 0020E10008000000:error:80000027:system library:dgram_sendmmsg:Destination address required:crypto/bio/bss_dgram.c:1314:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_sendmmsg:local address not available:crypto/bio/bss_dgram.c:1297:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_recvmmsg:local address not available:crypto/bio/bss_dgram.c:1498:
        # 0020E10008000000:error:1000006F:BIO routines:dgram_recvmmsg:local address not available:crypto/bio/bss_dgram.c:1527:
        # OPENSSL_TEST_RAND_ORDER=1662544797
        not ok 4 - iteration 4
# ------------------------------------------------------------------------------
    # OPENSSL_TEST_RAND_ORDER=1662544797
    not ok 1 - test_bio_dgram
# ------------------------------------------------------------------------------
../../util/wrap.pl ../../test/bio_dgram_test => 1
not ok 1 - running bio_dgram_test
# ------------------------------------------------------------------------------
#   Failed test 'running bio_dgram_test'
#   at /var/lib/buildbot/worker/ossl-freebsd13-x86_64/master_unix-freebsd13-x86_64/build/util/perl/OpenSSL/Test/Simple.pm line 77.
# Looks like you failed 1 test of 1.04-test_bio_dgram.t ................ 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
````

**Comment by mattcaswell** [comment]
@hlandau

**Comment by hlandau** [comment]
I'll investigate this.

**Comment by hlandau** [comment]
Fix in #19164.
